File: Download PDFs AT e SS/Autenticacao.cs
using OpenQA.Selenium;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace Download_PDFs_AT_e_SS
{
    class Autenticacao
    {
        internal static void AutenticarAT(IWebDriver driver, Empresa empresa)
        {
            driver.Navigate().GoToUrl("https://www.acesso.gov.pt/v2/loginForm?partID=PFAP&path=/geral/dashboard");
            Thread.Sleep(500);
            
            driver.FindElement(By.XPath("//*[@id=\"radix-:ra:-trigger-N\"]")).Click();

            driver.FindElement(By.Name("username")).SendKeys(empresa.NIF);
            driver.FindElement(By.Name("password")).SendKeys(empresa.PasswordAT);
            Thread.Sleep(500);
            driver.FindElement(By.XPath("/html/body/div/div/main/div[2]/div[2]/div[1]/div[3]/form/button")).Click();
        }

        internal static void AutenticarFundosCompensacao(IWebDriver driver, Empresa empresa)
        {
            driver.Navigate().GoToUrl("https://www.fundoscompensacao.pt/sso/login?service=https%3A%2F%2Fwww.fundoscompensacao.pt%2Ffc%2Fcaslogin");
            Thread.Sleep(500);
            driver.FindElement(By.Id("toogleAuth")).Click();
            driver.FindElement(By.Id("username")).SendKeys(empresa.NISS);
            driver.FindElement(By.Id("password")).SendKeys(empresa.PasswordSS);
            Thread.Sleep(500);
            driver.FindElement(By.Id("submitBtn")).Click();

            Thread.Sleep(500);
            //if continuarBtn exists
            var continuarBtnArray = driver.FindElements(By.Id("continuarBtn"));
            if (continuarBtnArray.Count > 0)
            {
                continuarBtnArray[0].Click();
            }
        }

        internal static void AutenticarSS(IWebDriver driver, Empresa empresa)
        {
            driver.Navigate().GoToUrl("https://www.seg-social.pt/sso/login?service=https%3A%2F%2Fwww.seg-social.pt%2Fptss%2Fcaslogin");
            Thread.Sleep(500);
            driver.FindElement(By.Id("toogleAuth")).Click();
            driver.FindElement(By.Id("username")).SendKeys(empresa.NISS);
            driver.FindElement(By.Id("password")).SendKeys(empresa.PasswordSS);
            Thread.Sleep(500);
            driver.FindElement(By.Id("submitBtn")).Click();

            Thread.Sleep(500);
            //if continuarBtn exists
            var continuarBtnArray = driver.FindElements(By.Id("continuarBtn"));
            if (continuarBtnArray.Count > 0)
            {
                continuarBtnArray[0].Click();
            }
        }
    }
}

File: novo/autenticacao.py
import time
from selenium.webdriver.common.by import By


class Autenticacao:
    @staticmethod
    def autenticar_at(driver, empresa):
        driver.get(
            "https://www.acesso.gov.pt/v2/loginForm?partID=PFAP&path=/geral/dashboard"
        )
        time.sleep(0.5)
        driver.find_element(By.XPATH, "//*[@id='radix-:ra:-trigger-N']").click()
        driver.find_element(By.NAME, "username").send_keys(empresa.nif)
        driver.find_element(By.NAME, "password").send_keys(empresa.password_at)
        time.sleep(0.5)
        driver.find_element(
            By.XPATH, "/html/body/div/div/main/div[2]/div[2]/div[1]/div[3]/form/button"
        ).click()

    @staticmethod
    def autenticar_fundos_compensacao(driver, empresa):
        if not empresa.password_ss:
            print(
                f"Empresa {empresa.nome} não tem password SS, não é possível autenticar nos Fundos de Compensação"
            )
            return

        driver.get(
            "https://www.fundoscompensacao.pt/sso/login?service=https%3A%2F%2Fwww.fundoscompensacao.pt%2Ffc%2Fcaslogin"
        )
        time.sleep(0.5)
        driver.find_element(By.ID, "toogleAuth").click()
        time.sleep(1)
        driver.find_element(By.ID, "username").send_keys(empresa.niss)
        time.sleep(0.5)
        driver.find_element(By.ID, "password").send_keys(empresa.password_ss)
        time.sleep(0.5)
        driver.find_element(By.ID, "submitBtn").click()
        time.sleep(0.5)
        continuar_btns = driver.find_elements(By.ID, "continuarBtn")
        if continuar_btns:
            continuar_btns[0].click()

    @staticmethod
    def autenticar_ss(driver, empresa):
        driver.get(
            "https://www.seg-social.pt/sso/login?service=https%3A%2F%2Fwww.seg-social.pt%2Fptss%2Fcaslogin"
        )
        time.sleep(0.5)
        driver.find_element(By.ID, "toogleAuth").click()
        time.sleep(1)
        driver.find_element(By.ID, "username").send_keys(empresa.niss)
        time.sleep(0.5)
        driver.find_element(By.ID, "password").send_keys(empresa.password_ss)
        time.sleep(0.5)
        driver.find_element(By.ID, "submitBtn").click()
        time.sleep(0.5)
        continuar_btns = driver.find_elements(By.ID, "continuarBtn")
        if continuar_btns:
            continuar_btns[0].click()
